STEPS.md

```diff
@@ -0,0 +1,195 @@
+# How to build Awesome Tasks, step by step
+
+This document describes how to build Awesome Tasks step by step.
+We will be introducing new concepts on each step, so make sure you understand what's going on before going further! If you have any doubts, please ask your coach for help!
+
+## Step 1
+### From basic HTML markup and styling to a dynamic list of tasks
+
+We will start with a new HTML file called `index.html`. We will use it as the skeleton of our little app. Here is how it will look like:
+
+*1.1 —`index.html` base markup*
+```html
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>Awesome Tasks</title>
+  </head>
+  <body>
+    <h2>Awesome Tasks</h2>
+    <ul id="task-list">
+      <li>Buy coffee</li>
+      <li>Buy milk</li>
+    </ul>
+  </body>
+</html>
+```
+
+If you open this file in a web browser, you will see a pretty boring HTML page that shows a title and a static list of things to do. Not that much, to be honest!
+
+The first thing we are going to do will be taking control of the task list: we will use JavaScript for adding a new task.
+
+Before `</body>`, let’s add this:
+
+*1.2 — The `<script>` tag*
+```html
+<script>
+  var listElement = document.querySelector('#task-list');
+  listElement.innerHTML = listElement.innerHTML + "<li>Disco dance</li>";
+</script>
+```
+
+If we reload the page now, we will see a new task popping up in our list: it comes straight from JavaScript!
+
+**What we just did:**
+  * We used the `document.querySelector()` DOM API to access our `#task-list` element
+  * We modified its `innerHTML` property by adding some more HTML: our new task item.
+
+## Step 2
+### All the tasks from JavaScript
+
+Now we know how to use JavaScript to modify an element’s `innerHTML`.
+Let’s introduce a new concept: the `Array`. We will need this concept for grouping all our tasks and to make things dynamic.
+
+Let’s remove all our static tasks from the page:
+
+*2.1 — Let’s empty the `<ul>`*
+```html
+<ul id="task-list"></ul>
+```
+
+And now let's use an `Array` for display all our tasks
+
+*2.2 — All the tasks rendered from JavaScript*
+```html
+<script>
+  var listItems = ['Buy coffee', 'Buy milk', 'Disco dance'];
+  var listElement = document.querySelector('#task-list');
+  listItems.forEach(function(task) {
```

Comment:
I prefer higher-order functions myself, but I think that explaining looping and the concept of iteration in general deserves a little bit of attention. If they start to play around JS code the day after the course they'll meet a `for` loop sooner or later and it would be better if they knew what they are looking at (and why higher-order functions are sometimes preferable).